SleepEngine/Engine/Object.cpp

```diff
@@ -1,21 +1,79 @@
 #include "stdafx.h"
 #include "Object.h"
 #include <Engine/GameTemplate/Game.h>
+#include <Engine/Utils/stl_utils.h>
 
 BEGIN_NAMESPACE_SLEEP
 
-void Object::addComponent(ComponentsContainer::value_type&& component)
+Component* Object::addComponent(ComponentsContainer::value_type&& component)
 {
     component->setParent(this);
+    Component* const handle = component.get();
     m_components.emplace_back(std::move(component));
+    return handle;
+}
+
+Object* Object::addChild(std::unique_ptr<Object>&& childPtr)

```

Comment:
please, don't put these methods in Object. Create a Parent component (or name it differently).